compiler/rustc_ast/src/ast.rs

```diff
@@ -1684,32 +1684,20 @@ pub enum AttrArgs {
     /// Delimited arguments: `#[attr()/[]/{}]`.
     Delimited(DelimArgs),
     /// Arguments of a key-value attribute: `#[attr = "value"]`.
-    Eq(
+    Eq {
```

Comment:
I don't see why we wouldn't do it in this PR. It would only get marginally smaller,  and this PR moves responsibilities from `ast::Attribute` to `hir::Attribute`, and a part of that is removing hir-parts from `ast::Attribute`. Unless you absolutely insist, I'd argue including it right now isn't so bad.

Reply:
I've seen distracting changes across the PR just from refactoring `Eq(..)` into `Eq { .. }` alone, I'd prefer to skip the nonessential parts.

Reply:
I disagree with this. The whole point of this PR is to remove Eq, the rest of the PR is simply all the infrastructure around that to enable it to be removed. If we don't remove it, it literally becomes a variant that is unused. We never use it because we lower to hir attributes. I think it's important to keep it in.

Reply:
I've split this up into https://github.com/rust-lang/rust/pull/133746, https://github.com/rust-lang/rust/pull/131808/commits/b61c7c4478f57912354db74fd444fec3bac85e4b and https://github.com/rust-lang/rust/pull/131808/commits/340edd0af1757e989daf3b32265eeeb45d6bd30d